### DISCUSSION TITLE
How to pragmatically get authentication object in any method ?

### DISCUSSION THREAD

**Discussion by siddharth223437** [micronaut-projects/micronaut-security#1015] [Q&A] [answered]
How to pragmatically get authentication object in any method ?

In springboot i used

securitycontextholder.getcontext().getauthentication()

**Comment by sdelamo** [comment, marked as answer]
See: https://micronaut-projects.github.io/micronaut-security/latest/guide/#securityService

> 3.1 User outside of a controller

> If you need to access the currently authenticated user outside of a controller, you can inject [SecurityService](https://micronaut-projects.github.io/micronaut-security/latest/api/io/micronaut/security/utils/SecurityService.html) bean, which provides a set of convenient methods related to authentication and authorization.
